ports/geckolib/glue.rs

```diff
@@ -410,3 +411,48 @@ pub extern "C" fn Servo_DropStyleSet(data: *mut RawServoStyleSet) -> () {
         let _ = Box::<PerDocumentStyleData>::from_raw(data as *mut PerDocumentStyleData);
     }
 }
+
+pub struct GeckoDeclarationBlock {
+    pub declarations: Option<PropertyDeclarationBlock>,
+    pub cache: *mut nsHTMLCSSStyleSheet,
+    pub immutable: bool,
+}
+
+#[no_mangle]
+pub extern "C" fn Servo_ParseStyleAttribute(bytes: *const u8, length: u32,
+                                            cache: *mut nsHTMLCSSStyleSheet)
+                                            -> *mut ServoDeclarationBlock {
+    let value = unsafe { from_utf8_unchecked(slice::from_raw_parts(bytes, length as usize)) };
+    let declarations = box GeckoDeclarationBlock {
+        declarations: GeckoElement::parse_style_attribute(value),
+        cache: cache,
+        immutable: false,
+    };
+    Box::into_raw(declarations) as *mut ServoDeclarationBlock
+}
+
+#[no_mangle]
+pub extern "C" fn Servo_DropDeclarationBlock(declarations: *mut ServoDeclarationBlock) {
+    unsafe {
+        let _ = Box::<GeckoDeclarationBlock>::from_raw(declarations as *mut GeckoDeclarationBlock);
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn Servo_GetDeclarationBlockCache(declarations: *mut ServoDeclarationBlock)
+                                                 -> *mut nsHTMLCSSStyleSheet {
+    let declarations: &GeckoDeclarationBlock = unsafe { transmute(declarations) };
```

Comment:
We should be able to remove these transmutes by mapping the types together and then using .as_ref(). We do this elsewhere IIRC.